utilities/src/serialize/helpers.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// The snarkVM library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkVM library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkVM library. If not, see <https://www.gnu.org/licenses/>.
+
+pub use crate::{
+    io::{self, Read, Write},
+    FromBytes,
+    ToBytes,
+    Vec,
+};
+use crate::{serialize::traits::*, SerializationError};
+
+/// Serialize a Vector's elements without serializing the Vector's length
+/// If you want to serialize the full Vector, use `CanonicalSerialize for Vec<T>`
+pub fn serialize_vec_without_len(
+    src: &Vec<impl CanonicalSerialize>,
```

Comment:
```suggestion
    src: &[impl CanonicalSerialize],
```

Reply:
and as always, an `Iterator` would be even better (as the body would allow it)

Reply:
c3bf5b26a